oauthproxy.go

```diff
@@ -1111,9 +1111,18 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 func isAjax(req *http.Request) bool {
 	acceptValues := req.Header.Values("Accept")
 	const ajaxReq = applicationJSON
+	// Iterate over multiple Accept headers, i.e.
+	// Accept: application/json
+	// Accept: text/plain
 	for _, v := range acceptValues {
-		if v == ajaxReq {
-			return true
+		// Iterate over multiple mimetypes in a single header, i.e.
+		// Accept: application/json, text/plain, */*
```

Comment:
The original intent of this was to figure out non-human driven requests that couldn't handle a redirect to a signin page (e.g. SPA backend API requests, machine to machine API auth). At least that's the only spot this function is used.

Are there scenarios where normal human driven interactions would have multiple headers (including `application/json`) and need the login redirect and don't want the `401 Unauthorized` a `true` from this would result in?

I don't know all the nuances of various browsers or client environments to answer this confidently.

Reply:
> Are there scenarios where normal human driven interactions would have multiple headers (including application/json) and need the login redirect and don't want the 401 Unauthorized a true from this would result in?

("multiple headers" use-case is covered in current code, what you probably think about is "multiple mimetypes in one header" case, which is what I've extended) 

I don't think so, but I might be missing some edge-case too. However, XHR detection is failing right now with clients using multiple mimetypes by default (e.g. Axios, which is still extremely popular). The reasoning for multiple mimetypes is probably a kind of catch-all -- think about APIs working with JSON, but occasionally returning images. I honestly can't imagine breaking someone's use case, since they'd be knowingly relying on buggy behavior, and this patch corrects what has been intended, since finding `application/json` in `Accept` headers has been the intention of `isAjax` all along.

PS. BTW, `*/*` would match `application/json` too, but that's a bit far-fetched as an XHR indicator.

Reply:
Good point, I completely missed in the diff the multi-acceptValues was already supported. So this is likely fine as well 👍